### ISSUE TITLE
Discrepency in reported "initial size" and asset list in v5.25.1+

### ISSUE THREAD

**Issue by hyshka** [statoscope/statoscope#213]
**Describe the bug**

I have a number of complex Webpack entry points and found some discrepancies between Statoscope v5.24.0 and v5.25.1. I'm assuming this was introduced in v5.25.0 but that version failed to generate a report due to an error that was fixed in v5.25.1. This issue is also present in the latest v5.27.0 release.

After some digging, I found that in v5.25.1+ some of our entry points:
- The "initial size" was larger.
- The list of "assets" is greater than the list of "chunks.

<img width="429" alt="image" src="https://github.com/statoscope/statoscope/assets/2090758/404690d7-76e7-4253-a9f7-a7b72a4cf084">

**To Reproduce**

I know this could be related to our complex setup but I wanted to check if this is a known issue before trying to reproduce this in a minimal example.

**Expected behavior**

I would expect:
- The "initial size" should be more closely aligned with v5.24.0.
- The counts in "chunks" and "assets" should match as in v5.24.0.

<img width="409" alt="image" src="https://github.com/statoscope/statoscope/assets/2090758/e0f4188d-c811-48a2-b4e2-9a05114fd77d">

**Versions**:
 - Node (for non-browser bug): v16
 - Browser (for browser-bug): 
 - Webpack: v5.83.1
 - Statoscope: v5.25.1 (and greater)

**Additional context**

Let me know what else I could report to help narrow down the cause.

**Comment by smelukov** [comment]
👋 Thanks for the issue!
In 5.25.1 I made some refactoring in assets info collecting to support some webpack 5 features. So, assets count and their size in a report should be more accuracy after that fix.
Could you provide some stats for more detailed investigation?

**Comment by hyshka** [comment]
@smelukov Yes, that seems like it could be related. I know it's possible that this could be more accurate than what it was before but I'm not sure how to validate that on my end.

Here's a stats file with v5.25.1 that will be accessible for a week: https://drive.proton.me/urls/PG86RGSSTG#Qv0cDs6oiAyg

**Comment by smelukov** [comment]
@hyshka could you check what exactly assets are injecting into a page with presentation_edit entrypoint?

**Comment by hyshka** [comment]
@smelukov This is interesting. The total number of assets on the page is 28. It is complicated by using [django-webpack-loader](https://github.com/django-webpack/django-webpack-loader) and including a global entry point. Two entry points are included on this page but django-webpack-loader skips any common assets so they're not repeated in the DOM.

For the global muckrack entry point, 13 assets:
```
runtime.496dbe4066c600d04713.bundle.js
6468.510dde9a19aa05be3c6f.bundle.js
3232.cca65ea969bd42e9ecbb.bundle.js
9755.3630453b2ce6597fc918.bundle.js
7658.3f0281cb354fee72c0f4.bundle.js
942.e432dbf35c257f6614f9.bundle.js
5659.7df03be816ff5123d2c7.bundle.js
2718.662f6ec2f96219d00498.bundle.js
497.9c3e08dad4a5bdad551e.bundle.js
1817.8dd7c8ccc11cb90eb0c8.bundle.js
4166.1436de74687c4c35a19b.bundle.js
4618.6636c87f049afc647a58.bundle.js
muckrack.e7affe34155150b7cd4f.bundle.js
```

For the `presentation_edit` entry point, 15 assets:
```
7695.6f85b04628b076f15ba4.bundle.js
1921.654e92e82a83988e96d1.bundle.js
7597.7b96318420f753495277.bundle.js
8840.aa8cbae7cd5b45e16e63.bundle.js
7844.167bea28973dacc51497.bundle.js
178.7ce2d8e58800d5b07e8c.bundle.js
6950.2c1026e6d78b97b45fd8.bundle.js
6095.4a73c489b2cf13560880.bundle.js
3465.1bb4f41c50a43b756f11.bundle.js
2706.5473089414da535aed9b.bundle.js
8714.78edc768ac7c8afb9022.bundle.js
9306.59e345fcfcad99c284bb.bundle.js
3245.1b25e591125e25e00920.bundle.js
2539.860fc92f2449d19d08fb.bundle.js
presentations_edit.97e4201f95b4dc15d2e8.bundle.js
```

As far as I know, Webpack should not be aware that these entry points are used together so the full list of assets for the `presentation_edit` entry point should look different. If I were to remove the global entry point from this page, we'd see the following assets:

```
runtime.496dbe4066c600d04713.bundle.js
6468.510dde9a19aa05be3c6f.bundle.js
3232.cca65ea969bd42e9ecbb.bundle.js
9755.3630453b2ce6597fc918.bundle.js
7658.3f0281cb354fee72c0f4.bundle.js
942.e432dbf35c257f6614f9.bundle.js
7695.6f85b04628b076f15ba4.bundle.js
1921.654e92e82a83988e96d1.bundle.js
7597.7b96318420f753495277.bundle.js
8840.aa8cbae7cd5b45e16e63.bundle.js
7844.167bea28973dacc51497.bundle.js
178.7ce2d8e58800d5b07e8c.bundle.js
6950.2c1026e6d78b97b45fd8.bundle.js
6095.4a73c489b2cf13560880.bundle.js
3465.1bb4f41c50a43b756f11.bundle.js
2706.5473089414da535aed9b.bundle.js
8714.78edc768ac7c8afb9022.bundle.js
9306.59e345fcfcad99c284bb.bundle.js
3245.1b25e591125e25e00920.bundle.js
2539.860fc92f2449d19d08fb.bundle.js
presentations_edit.97e4201f95b4dc15d2e8.bundle.js
```

**Comment by smelukov** [comment]
@hyshka sorry for silence here, I was searching more similar cases that you described here and I've faced a similar problem in my project and seems like it should not work this way.
I'll rollback a part of 5.25.1 that changes assets collecting logic

**Comment by smelukov** [comment]
@hyshka fixed in 5.28.1

**Comment by hyshka** [comment]
@smelukov Thank you, my stats files now report the same hash between v5.24.0 and v5.28.1. There are no differences between build now.
